lib/src/dv360_service_response_parser.dart

```diff
@@ -36,10 +36,37 @@ class InsertionOrderParser {
       key: (v) => v.name,
       value: (v) => v);
 
-  /// Parses a json string to [InsertionOrder].
-  static InsertionOrder parse(String jsonString) {
-    Map<String, dynamic> map = json.decode(jsonString);
-    return _createInsertionOrder(map);
+  /// Parses a list of [InsertionOrder] from a [jsonString] .
+  static List<InsertionOrder> parseInsertionOrders(String jsonString) {
+    Map<String, dynamic> map;
+    try {
+      map = json.decode(jsonString);
+    } catch (e) {
+      // [jsonString] is not in valid json format, returns empty list.
+      print('invalid json format');
+      return [];
+    }
+
+    return map.containsKey('insertionOrders')
+        ? List.from(map['insertionOrders'])
+            .map((ioMap) => _createInsertionOrder(ioMap))
+            .toList()
+        : [_createInsertionOrder(map)];
+  }
+
+  /// Parses the nextPageToken from a [jsonString].
+  ///
+  /// Returns empty string is input [jsonString] is null or empty.
+  static String parseNextPageToken(String jsonString) {
+    Map<String, dynamic> map;
+    try {
+      map = json.decode(jsonString);
+    } catch (e) {
+      // [jsonString] is not in valid json format, returns empty string.
```

Comment:
In real cases, no. And in component testing with mocks, null will be passed into this function.
I guess sanity check and return empty string will be better then.

Reply:
let's keep throwing. an error may be introduced and we wont notice especially because we dont have a logging framework here. later we can have some top level error handling to fail more gracefully.